internal/controller/shared.go

```diff
@@ -108,19 +109,7 @@ func makeChildResourceFromRolloutAndUpdateSpecHash(
 		}
 	}
 
-	setAnnotation(rolloutObj, apiv1.KeyHash, childResouceSpecHash)
 	obj.Spec = childResourceSpec
 
-	return &obj, rolloutChildOp, nil
-}
-
-func setAnnotation(obj metav1.Object, key, value string) {
-	annotations := obj.GetAnnotations()
-	if annotations == nil {
-		annotations = make(map[string]string)
-	}
-
-	annotations[key] = value
-
-	obj.SetAnnotations(annotations)
+	return &obj, rolloutChildOp, childResouceSpecHash, nil
```

Comment:
`if annotation, exists := annotations[apiv1.KeyHash]; exists && annotation != childResouceSpecHash {`

I missed this earlier. But looking at this line above, what if `exists==false` above?